apps/omg/lib/omg/ethereum_event_listener/core.ex

```diff
@@ -138,6 +144,7 @@ defmodule OMG.EthereumEventListener.Core do
 
     height_to_check_in = get_height_to_check_in(new_state)
     db_update = [{:put, update_key, height_to_check_in}]
+    :ok = :telemetry.execute([:process, __MODULE__], %{events: events}, state)
```

Comment:
can we call this outside of `.Core` so that it remains side-effect free?

Reply:
hm... what causes the side effect?

Reply:
Isn't talking to `:telemetry` one? 

Sorry if I'm having false assumptions here, but I'm guessing this talks to some dependency to do sth. `.Core.xyz` functions are written with the intent of remaining pure. 

Unless absolutely necessary we avoid doing calling external stuff in `.Core.xyz`, I think the only exception was made for `_ = Logger...` calls, but only in a few cases.

`events` and `state` are available outside, so the move seems to be an easy one.

Reply:
This event gets forwarded to the dedicated Ethereum event listener handler located in module `OMG.EthereumEventListener.Measure`

I don't quite understand what you mean by _side effect_ though indeed the change isn't big, but would like to understand where `keeping .Core.xyz` pure comes from.

Reply:
> would like to understand where keeping .Core.xyz pure comes from.

Sure thing! It comes from the approach adopted, so segregate logic into such `Core` modules, with no side effects/side causes. These modules would hold pure functions (same output for same input etc.). Then they can be robustly/quickly tested in unit tests and also I think they are easier to reason about, in general, being just these reliable chunks of logic.